src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAssignableTypeForForeachVariable.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class UseAssignableTypeForForeachVariable : DiagnosticAnalyzer
```

Comment:
I don't know if this is applicable to VB, but noting in case if it's forgotten.

Reply:
This is C# only analyzer for now.

Reply:
@Evangelink @mavasani Do we want a tracking issue to follow-up with VB?

Reply:
See #4875